### DISCUSSION TITLE
Format value in <input> when out of focus

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#8104] [Q&A]
We need to display values in unfocused input fields formatted with [Number.toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) while keeping the original raw value during editing/focus à la MS Excel.

![c7a07475-0778-4b5f-8d80-69d99a1ad7a7](https://user-images.githubusercontent.com/4553604/160427636-a6357c40-5807-4dc0-adc3-833c8b81045d.gif)

How do I achieve this without cheating RHF? I don't want to get the formatted value back to me in the submit handler.

**Comment by Moshyfawn** [comment]
It's not really a RHF question.

> I don't want to get the formatted value back to me in the submit handler

If you don't want to store the formatted value in RHF, you can use the React state and update it `onFocus` along with some `isFocused` state that would toggle your field's input display.
